velox/functions/lib/aggregates/noisy_aggregation/sketch/BitMap.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/common/base/IOUtils.h"
+#include "velox/common/memory/HashStringAllocator.h"
+
+namespace facebook::velox::functions::aggregate {
+
+using Allocator = facebook::velox::StlAllocator<int8_t>;
+/// A C++ implementation of the Bitmap class used in SfmSketch.
+/// This is a wrapper around std::vector<int64_t> which provides functionality
+/// similar to Java's BitSet.
+class BitMap {
+ public:
+  explicit BitMap(uint64_t length, HashStringAllocator* allocator)
+      : length_(length), bits_(Allocator(allocator)) {
+    bits_.resize(velox::bits::nbytes(length_));
+  }
+
+  bool getBit(uint64_t position) const {
+    return bits::isBitSet(bits_.data(), position);
+  }
+
+  void setBit(uint64_t position, bool value) {
+    bits::setBit(bits_.data(), position, value);
+  }
+
+  void flipBit(uint64_t position) {
+    auto bitsAs8Bit = reinterpret_cast<uint8_t*>(bits_.data());
+    bitsAs8Bit[position / 8] ^= 1 << (position % 8);
+  }
+
+  uint64_t length() const {
+    return length_;
+  }
+
+  // Count the number of bits set to 1.
+  uint64_t bitCount() const {
```

Comment:
Are there no suitable utilities in bits namespace?

Move non-trivial implementations to .cpp